qcodes/instrument/visa.py

```diff
@@ -158,10 +160,12 @@ def set_terminator(self, terminator: str) -> None:
         if self.visabackend == 'sim':
             self.visa_handle.write_termination = terminator
 
-    def _set_visa_timeout(self, timeout: Optional[Union[float, int]]) -> None:
-
+    def _set_visa_timeout(self, timeout: Optional[float]) -> None:
+        # according to https://pyvisa.readthedocs.io/en/latest/introduction/resources.html#timeout
```

Comment:
None is still supported but since None is never returned I typed `timeout` using only `Attribute[float]`. I could type it as `Attribute[Optional[float]]` but that would lose some precision on the returned type.